### PR TITLE
Use correct thread stack limits in jl_init_stack_limits

### PR DESCRIPTION
**Pull request by fingolfin**
Actually these do not really seem to matter much in practice (as explained to me by @vtjnash on Slack, thank you). The one thing that did cause me some grief here, though, was that stack_hi was the low address and stack_lo the high address.

A smaller change to achieve that "hi means high, lo means low" would be to just reverse `stack_hi` and `stack_lo`.

### CODE REVIEW

File: src/init.c [hunk 1]
```diff
@@ -71,21 +71,17 @@ void jl_init_stack_limits(int ismaster, void **stack_lo, void **stack_hi)
...
         return;
 #  elif defined(_OS_DARWIN_)
         extern void *pthread_get_stackaddr_np(pthread_t thread);
         extern size_t pthread_get_stacksize_np(pthread_t thread);
         pthread_t thread = pthread_self();
         void *stackaddr = pthread_get_stackaddr_np(thread);
-        *stack_lo = (void*)stackaddr;
-        *stack_hi = (void*)__builtin_frame_address(0);
+        size_t stacksize = pthread_get_stacksize_np(thread);
+        *stack_hi = stackaddr;
```
Comment (vtjnash) [inline src/init.c:83]: Don't you meant to switch these, since the malloc region is stackaddr to stackaddr+stacksize?

Comment (fingolfin) [reply src/init.c:83]: Maybe I am misunderstanding you (sorry), but: stacks on all OS/CPU combos supported by Julia grow downwards, the stack starts at the highest address `stackaddr` and grows downwards towards `stackaddr-stacksize` (right after which the guard pages comes by default, at least for pthread created stacks with default settings).

I double checked this in darwin_libpthread, glibc and muslc and via real-world experiments on darwin/macos and linux+glibc (no muslc for me

Comment (fingolfin) [reply src/init.c:83]: @vtjnash does my explanation make sense to you? Any other questions?

Comment (vtjnash) [reply src/init.c:83]: Okay, yeah, it is confusing that it is inverted from the values returned by `pthread_attr_getstack`, but apparently that is the legacy behavior

**Pull request merged**